pkg/localkube/storage_provisioner.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package localkube
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/external-storage/lib/controller"
+	"github.com/kubernetes-incubator/external-storage/lib/leaderelection"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/types"
+	"k8s.io/client-go/pkg/util/uuid"
+	"k8s.io/client-go/pkg/util/wait"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	resyncPeriod              = 15 * time.Second
+	provisionerName           = "k8s.io/minikube-hostpath"
+	exponentialBackOffOnError = false
+	failedRetryThreshold      = 5
+	leasePeriod               = leaderelection.DefaultLeaseDuration
+	retryPeriod               = leaderelection.DefaultRetryPeriod
+	renewDeadline             = leaderelection.DefaultRenewDeadline
+	termLimit                 = leaderelection.DefaultTermLimit
+)
+
+type hostPathProvisioner struct {
+	// The directory to create PV-backing directories in
+	pvDir string
+
+	// Identity of this hostPathProvisioner, generated. Used to identify "this"
+	// provisioner's PVs.
+	identity types.UID
+}
+
+func NewHostPathProvisioner() controller.Provisioner {
+	return &hostPathProvisioner{
+		pvDir:    "/tmp/hostpath-provisioner",
```

Comment:
That's actually a different path. I'll add both, I considered switching this to use that same path, but I'd rather avoid having two provisioners running in the same directory.